sdk/src/main/scala/io/horizen/wallet/AbstractWallet.scala

```diff
@@ -116,19 +116,16 @@ abstract class AbstractWallet[
       case None => 0
     }
     for (_ <- 0 to secretsNumber) {
+      nonce += 1
```

Comment:
I think we are introducing a non backward compatible change here
the first secret must be generated with nounce = 0 , while now we are always skipping it.
change this line with:
if (nonce > 0) { nonce += 1}

Reply:
Done

Reply:
sorry I realized we are inside a for loop  and if nonce = 0 and  secrtestNumber is > 1 this soultion does not work
let's try to rearrange the code

Reply:
Done
If condition put before the for loop and nonce incremented at the end of the loop